tensorflow_model_optimization/g3doc/guide/pruning/pruning_with_sparsity_2_by_4.ipynb

```diff
@@ -0,0 +1,597 @@
+        "weights_to_display = weights_to_display[0:width, 0:height]\n",
+        "\n",
+        "val_ones = np.ones([height, width])\n",
+        "val_zeros = np.zeros([height, width])\n",
+        "subset_values_to_display = np.where(abs(weights_to_display) > 1e-9, val_ones, val_zeros)\n",
+        "\n",
+        "plot_separation_lines(height, width)\n",
+        "\n",
+        "plt.axis('off')\n",
+        "plt.imshow(subset_values_to_display)\n",
+        "plt.colorbar()\n",
+        "plt.title(\"Structurally pruned weights for Conv2D layer\")\n",
+        "plt.show()"
+      ],
+      "outputs": [],
+      "metadata": {}
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Let's see how randomly pruned weights look. We extract them and display a subset of the weight tensor."
+      ],
+      "metadata": {}
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "source": [
+        "# Let us get weights of the convolutional layer that has been pruned with random pruning.\n",
+        "tensor_name = 'pruning_sparsity_0_5/Conv2D'\n",
+        "detail = [x for x in details if tensor_name in x[\"name\"]]\n",
+        "tensor_data = interpreter.tensor(detail[0][\"index\"])()\n",
+        "print(f\"Shape of the weight tensor is {tensor_data.shape}\")"
+      ],
+      "outputs": [],
+      "metadata": {}
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "source": [
+        "weights_to_display = tf.reshape(tensor_data, [tensor_data.shape[0],tf.reduce_prod(tensor_data.shape[1:])])\n",
+        "weights_to_display = weights_to_display[0:width, 0:height]\n",
+        "\n",
+        "val_ones = np.ones([height, width])\n",
+        "val_zeros = np.zeros([height, width])\n",
+        "subset_values_to_display = np.where(abs(weights_to_display) > 0, val_ones, val_zeros)\n",
+        "\n",
+        "plot_separation_lines(height, width)\n",
+        "\n",
+        "plt.axis('off')\n",
+        "plt.imshow(subset_values_to_display)\n",
+        "plt.colorbar()\n",
+        "plt.title(\"Unstructed pruned weights for Conv2D layer\")\n",
+        "plt.show()"
+      ],
+      "outputs": [],
+      "metadata": {}
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "There is a python script included in the TensorFlow Model Optimization Toolkit that could be used to check whether which layers in the model from the given flite file have the structurally pruned weights: [`check_sparsity_m_by_n.py`](https://github.com/tensorflow/model-optimization/blob/master/tensorflow_model_optimization/python/core/sparsity/keras/tools/check_sparsity_m_by_n.py). The usage of this tool for the case of 2 by 4 is shown below:"
+      ],
+      "metadata": {}
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "`python ./tensorflow_model_optimization/python/core/sparsity/keras/tools/check_sparsity_m_by_n.py --model_tflite=pruned_model.tflite --m_by_n=2,4`"
```

Comment:
Could you format it to a shell script as you did above in Setup section? e.g, "! pip install ... "